### DISCUSSION TITLE
Is it possible to use DeePMD-kit official Docker image as base image while recompiling Lammps?

### DISCUSSION THREAD

**Discussion by Cloudac7** [deepmodeling/deepmd-kit#2568] [Q&A]
I am now working on how to recompile a specific Lammps version supporting Plugin. As I just expected, it should be a seamless work just using `ghcr.io/deepmodeling/deepmd-kit:2.2.1_cuda11.6_gpu` as base image to recompile the Lammps code in need in shared version. However, it raised something like this:

```
  using 2 OpenMP thread(s) per MPI task
Open of file /opt/deepmd-kit/lib/deepmd_lmp/dpplugin.so failed: /opt/deepmd-kit/lib/deepmd_lmp/dpplugin.so: undefined symbol: _ZN9LAMMPS_NS5Error4_allERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEiN3fmt6v8_lmp17basic_string_viewIcEENSA_17basic_format_argsINSA_20basic_format_contextINSA_8appenderEcEEEE
Loaded 0 plugins from /opt/deepmd-kit/lib/deepmd_lmp
```

In my understanding, there might be something wrong with the namespace `LAMMPS_NS` and I thus moved on to try to compile DeePMD-kit C++ interface, while it could not find CUDA environment even if I manually point it to what in `/usr/local/cuda`. I just found a few runtime libraries lacking in compiling support.

So how could I just recompile Lammps enabling the compiled `dpplugin.so` or recompile it together?

**Comment by njzjz** [comment]
Please check #2523, which reported the same error.

**Reply by Cloudac7** [reply]
I just tried to add `-DCMAKE_CXX_STANDARD=17` when compiling Lammps and also tried to use the latest 2.2.2 version as decribed in the discussion mentioned. However, it still raised the error above.

I also tried to just install and build a clean DeePMD-kit C++ Interface with `nvidia/cuda:11.7.1-cudnn8-devel-ubuntu20.04` image and copied the built `/opt/deepmd-kit` to the former built mirror. It raised that:

```
Open of file /opt/deepmd-kit/lib/deepmd_lmp/dpplugin.so failed: /opt/deepmd-kit/lib/libdeepmd_cc.so: undefined symbol: _ZN3tsl3Env7DefaultEv
```

**Reply by njzjz** [reply]
I strongly recommend the new way in v2.2.2, which builds from the C library and avoid liking the C++ symbols.
See https://docs.deepmodeling.com/projects/deepmd/en/v2.2.2/install/install-from-c-library.html for details.
